### ISSUE TITLE
Open-source thank you

### ISSUE THREAD

**Issue by sjuxax** [NEZNAMY/TAB#200]
Hi @NEZNAMY,

Just wanted to make a public thank-you for open-sourcing this extension, and note that I bought a license solely in order to express my appreciation for that.

While I know it's not much, I hope this plays some small part in addressing the concern expressed in the README that you don't see any reason to open-source. You got at least one sale from it. 👍

I'm sending this as a GitHub issue rather than an email for visibility; maybe some other people will be inspired to demonstrate their appreciation similarly. The price is only ~$5 (+/- about $1.50, depending on the day's exchange rate). That's less than the price of a typical stop at the coffee shop or burger joint. Quite an affordable way to express support for both a great plugin and the open-source community.

Thanks
Jeff

**Comment by NEZNAMY** [comment]
Thank you for showing your appreciation. May i know how did the source code help you (in comparison to just decompiling the jar) ? I am trying to add features and fix bugs so no one should need to fork it to keep using it.

**Comment by AryToNeX** [comment]
Maybe you'd want to remove the transaction ID from the GitHub issue.

> May i know how did the source code help you (in comparison to just decompiling the jar)?

Decompiling the jar file means that we lose the ability to look up at comments (if available) and things will obviously be slightly different from the actual source code. Not to mention that, depending on the "quality" of the decompiler, some parts of the plugin may not be decompiled at all.

I would also add that contributing to the actual source code is another way to demonstrate appreciation for this project (sure, we all know that money is money and money is better than code as it can pay bills, but the community value behind open source projects with more than one contributor is also a thing).

(I soon will be a free TAB user, it seems good for me even in its free release; for now I am just looking at the code)

**Comment by NEZNAMY** [comment]
You just inspired me to comment the code (more). Currently I'm trying to name methods and classes the way you immediately know what they do, but sometimes it's really tricky. You can expect a lot more once I push the 2.8.5 release (already made 60 commits since the last release 2.8.4).

**Comment by sjuxax** [comment]
> Maybe you'd want to remove the transaction ID from the GitHub issue.

Yeah, I went back and forth on it, but thought it was important to demonstrate authenticity. That's done so I'll take it out now. Do you know of any specific attack vectors that exist with a transaction ID or is it just theoretical? It seems like the system would be very weak if any damage could be caused with that.

**Comment by AryToNeX** [comment]
> You just inspired me to comment the code (more). Currently I'm trying to name methods and classes the way you immediately know what they do, but sometimes it's really tricky. You can expect a lot more once I push the 2.8.5 release (already made 60 commits since the last release 2.8.4).

That's awesome!

> I'm still wondering what made you check out this plugin instead of many other open source plugins with the same features (or plugin combinations as this one contains features of several plugins).

Do you know PlayerListPlus? Once it became PlayerListDeluxe it was WAY TOO CONFUSING for me to use it. I switched to this one instead.

**Comment by AryToNeX** [comment]
> Do you know of any specific attack vectors that exist with a transaction ID or is it just theoretical? It seems like the system would be very weak if any damage could be caused with that.

@sjuxax I don't know any specific attack vector, but it's better to be safe than sorry anyways.

**Comment by NEZNAMY** [comment]
> _(pls opensource myskin next so I can look at it even better)_
This will not happen, as it's not even my plugin. It belongs to a friend who worked on SkinsRestorer before (massively reduces jar size, fixed and added many things), however he retired because the original author abandoned it and refused to transfer to the friend. Then he decided to make his own plugin and here it is. As you can see, it's 100x smaller (100kb compared to SR's 10MB). However, he kept getting negative feedback mostly from trolls and didn't see a reason to continue development, so he sent me source code, closed his account and transferred the resource to my account without me agreeing to it. Now I have it on my account and have no idea what to do with it.

**Comment by AryToNeX** [comment]
@NEZNAMY this is beyond fucked up, to say the least. Well, it might be the end of an otherwise really good plugin (and I say it because I use its API too in my toolbox and it just works)..

Since sharing sources is a no-go, I guess there isn't much to be done, right? But SR is a nonsensical bloat to me; finding a replacement is going to be tricky now.

By the way, a suggestion from me would be to update the plugin to support 1.16.2 by basically copying over the work made in 1.16.1 (safe guess that nothing really has changed in terms of packet sending and forgery) and then discontinue that project if you don't feel like continuing it.

Do you have any suggestions for replacements, btw?

**Comment by NEZNAMY** [comment]
I'll ask the original author if he can add 1.16.2 support. I have no idea what could be used as a replacement. Majority of owners just go for SR, which is another reason why he quit - there's a much more popular plugin out there which people prefer.

**Comment by AryToNeX** [comment]
Can you ask the original author to open source the plugin? It'll be very welcome.

**Comment by NEZNAMY** [comment]
I don't think he will want to. We talked about MC stuff and he wants to have nothing in common with it.

**Comment by xknat** [comment]
> > _(pls opensource myskin next so I can look at it even better)_
> 
> This will not happen, as it's not even my plugin. It belongs to a friend who worked on SkinsRestorer before (massively reduced jar size, fixed and added many things), however he retired because the original author abandoned it and refused to transfer to the friend. Then he decided to make his own plugin and here it is. Since then, SR just kept growing in size for no reason at all. As you can see, myskin is 100x smaller (100kb compared to SR's 10MB). However, he kept getting negative feedback mostly from trolls and didn't see a reason to continue development, so he sent me source code, closed his account and transferred the resource to my account without me agreeing to it. Now I have it on my account and have no idea what to do with it.

Hey there!
Just a late follow up on this, sorry.
The original author of SR was very busy and did not want to transfer the project.
I can agree that the original author of skinsrestorer did not put as much credits and faith in your friend after all he did.
Looking back to all of the changes and commits, i can agree that we would not have been where skinsrestorer stands now without your friend input. I do honor him for his additions to SR, and put his name on both our spigot page and author of our sponge plugin.
Having trolls is never fun, and it can be very demotivating when a project you worked hard on grow's very big and you end up being cast aside....
Sadly, things can't be turned back in time. We can only move forward.
Also, I don't see other skin plugins as competitors. I see them as more options for server owners to chose from :) 
Regards,
Knat SkinsRestorer

**Comment by BomBardyGamer** [comment]
I know this is kinda old now, but I thought I'd at least say my piece.

I've spent the past months working on Minecraft server software from scratch, and what I have found out is the hardest thing is getting people to use it, and the best way to do that is by expanding the ecosystem. Without plugins like this being open-source, I would be much more limited in what I could provide to users to encourage them to use the software. I'd really like to say my thank you to making this open-source, and allowing me to expand this to my ecosystem.

Bardy out
